### PR TITLE
#35 network interface added to environment

### PR DESCRIPTION
**Pull request by ja6a**
CRUD operations on the network interface for an environment. Currently there is an issue with the deletion step of the acceptance test.

### CODE REVIEW

**Comment by pegerto** [comment]
+ require a doc update

File: skytap/resource_skytap_vm.go [hunk 1]
```diff
@@ -50,6 +50,37 @@ func resourceSkytapVM() *schema.Resource {
 				ValidateFunc: validation.NoZeroValues,
 				Computed:     true,
 			},
+
+			"network_interface": {
```
Comment (pegerto) [inline skytap/resource_skytap_vm.go]: schema.TypeList

File: skytap/resource_skytap_vm.go [hunk 2]
```diff
@@ -93,7 +124,69 @@ func resourceSkytapVMCreate(d *schema.ResourceData, meta interface{}) error {
...
+	return update(d, meta, true)
+}
+
+func addNetworkAdapters(d *schema.ResourceData, meta interface{}, vmID string) error {
+	if _, ok := d.GetOk("network_interface"); ok {
+
+		client := meta.(*SkytapClient).interfacesClient
+		ctx := meta.(*SkytapClient).StopContext
+
+		networkInterfaces := d.Get("network_interface").(*schema.Set)
```
Comment (pegerto) [inline skytap/resource_skytap_vm.go]: Use a flatten help method

File: skytap/resource_skytap_vm.go [hunk 3]
```diff
@@ -93,7 +124,69 @@ func resourceSkytapVMCreate(d *schema.ResourceData, meta interface{}) error {
...
+		for _, v := range networkInterfaces.List() {
```
Comment (pegerto) [inline skytap/resource_skytap_vm.go]: Is there a case for a partial update on the interfaces attribute?

**Comment by pegerto** [comment]
Colin Henry @jcherry this PR has been merge because of blocking to publish service, happy to capture any issue in a sparate ticket.

**Comment by jcherry** [comment]
> Colin Henry @jcherry this PR has been merge because of blocking to publish service, happy to capture any issue in a sparate ticket.

I'm guessing you have the wrong person in your mention

File: skytap/resource_skytap_vm.go [hunk 4]
```diff
@@ -50,6 +50,37 @@ func resourceSkytapVM() *schema.Resource {
...
+							ValidateFunc: validation.SingleIP(),
+							Computed:     true,
+						},
+						"hostname": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.NoZeroValues,
+							Computed:     true,
+						},
+					},
```
Comment (jchenry) [inline skytap/resource_skytap_vm.go]: You're missing the network this interface binds to.

**Comment by jchenry** [comment]
Environment? Published services should bind to the Virtual Machine.

**Comment by jchenry** [comment]
@jcherry they meant me, sorry about that.

File: skytap/resource_skytap_vm.go (hunk 4 above)
Comment (ja6a) [reply skytap/resource_skytap_vm.go]: The network is referenced via the network_id attribute.

Comment (jchenry) [reply skytap/resource_skytap_vm.go]: Sorry, I misinterpreted that into being the network interface id. in that case, we're missing the network interface id.

**Comment by pegerto** [comment]
Hey @jcherry sorry for the typo

File: skytap/resource_skytap_vm.go (hunk 4 above)
Comment (pegerto) [reply skytap/resource_skytap_vm.go]: network_id is one of the properties? not sure if I understand?

Comment (jchenry) [reply skytap/resource_skytap_vm.go]: there is the `network_id`, which should refer to the network the interface should bind to, but there is the network interface's id. the latter does not appear to be present.

Comment (pegerto) [reply skytap/resource_skytap_vm.go]: @jchenry  agree, the network_interface_id is missing, it is a r/o and not referenced by any resource. A definition will be useful to store the state of the interface, in the current implementation modifying the interface isn't allowed

**Pull request merged**